eva-server/src/test/java/uk/ac/ebi/eva/server/ws/IdentifierWSServerV2IntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.server.ws;
+
+import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantWithSamplesAndAnnotationsService;
+import uk.ac.ebi.eva.lib.Profiles;
+import uk.ac.ebi.eva.server.configuration.MongoRepositoryTestConfiguration;
+
+import java.util.List;
+
+import static com.lordofthejars.nosqlunit.mongodb.MongoDbRule.MongoDbRuleBuilder.newMongoDbRule;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Import(MongoRepositoryTestConfiguration.class)
+@UsingDataSet(locations = {
+        "/test-data/variants.json",
+        "/test-data/files.json",
+        "/test-data/annotations.json",
+        "/test-data/annotation_metadata.json"
+})
+@ActiveProfiles(Profiles.TEST_MONGO_FACTORY)
+public class IdentifierWSServerV2IntegrationTest {
+
+    private static final String TEST_DB = "test-db";
+
+    @Rule
+    public MongoDbRule mongoDbRule = newMongoDbRule().defaultSpringMongoDb(TEST_DB);
+
+    @Autowired
+    MongoDbFactory mongoDbFactory;
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @Test
+    public void testForExisting() {
+        String url = "/v2/identifiers/rs199692280?species=mmusculus&assembly=grcm38";
+        ResponseEntity<List<Variant>> response = restTemplate.exchange(url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<List<Variant>>() {
+                });
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertTrue(response.getBody().size() > 0);
```

Comment:
checking at least the important fields like chromosome and position and alleles would be nice. just in case we are returning an object will all fields as null.

Applies to the other test class too.

Reply:
Fixed :smiley: